### PR TITLE
New Channel Modal

### PR DESCRIPTION
**Pull request by dzucconi**
newchannelf

Includes the visibility selector in the edit channel modal, as well.

### CODE REVIEW

**Comment by dzucconi** [comment]
I think what I'll do is manually push this to staging prior to merging so @csherron can do a Q/A pass.

**Comment by dzucconi** [comment]
Minor issues that I'm punting on for now: 

* Focused input placeholders are gray because of some aggressive CSS in our legacy codebase; haven't pinpointed it yet.
* Modal window defaults to 90% height; if it's auto-sized the pulldown looks weird because we can't break-out of the modal window. (Not without some interesting fixed-position tracking solution)

File: components/tips/view.coffee [hunk 1]
```diff
@@ -21,13 +24,7 @@ module.exports = class TipView extends Backbone.View
...
     switch @model.get 'trigger'
       when 'new:channel'
-        # Opens the new channel dialog
-        $dropdown = $('.js-triggerable-channel-create')
-        $dropdown.addClass hoverClass = 'DropdownHover--hover'
-
-        # Closes it when the body is clicked
-        $('body').one 'click', (e) ->
-          $dropdown.removeClass hoverClass
+        openNewChannelModal()
```
Comment (broskoski) [inline components/tips/view.coffee]: Thank you!

File: components/logged_in_navigation/components/channel_create/index.styl [hunk 2]
```diff
@@ -0,0 +1,38 @@
+.NewChannel
+  display flex
+  align-items center
+  align-content center
+  justify-content center
+  padding 0 0.75em
+
```
Comment (broskoski) [inline components/logged_in_navigation/components/channel_create/index.styl]: Does this need a hover state? Wondering how to make it easier to let people to know to click.

File: react/components/ChannelVisibilityPulldown/components/ChannelVisibilityPulldownOption/index.js [hunk 3]
```diff
@@ -1,29 +1,52 @@
...
+
 const ChannelVisibilityPulldownOption = ({
-  mode, visibility, label, description, explanation,
+  mode, purpose, selected, visibility, label, description, explanation,
 }) => (
-  <div>
-    {mode === 'resting' &&
-      <Text color={`channel.${visibility}`}>
+  <Container>
+    {purpose === 'value' &&
```
Comment (broskoski) [inline react/components/ChannelVisibilityPulldown/components/ChannelVisibilityPulldownOption/index.js]: My purpose is value

File: components/logged_in_navigation/components/channel_create/index.styl (hunk 2 above)
Comment (dzucconi) [reply components/logged_in_navigation/components/channel_create/index.styl]: Sure

File: react/components/NewChannelForm/index.js [hunk 4]
```diff
@@ -36,7 +49,33 @@ export default class NewChannelForm extends Component {
...
+      title, description, visibility, visit_channel,
+    } = this.state;
+
+    const variables = { title, description, visibility };
+
+    this.setState({ mode: 'creating' });
+
+    return createChannel({ variables })
+      .then(({ data: { create_channel: { channel } } }) => {
+        if (visit_channel) {
```
Comment (broskoski) [inline react/components/NewChannelForm/index.js]: Cool

**Pull request merged**